src/main/java/ua/okwine/productexpirationdate/controller/SupplierController.java

```diff
@@ -0,0 +1,60 @@
+package ua.okwine.productexpirationdate.controller;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import ua.okwine.productexpirationdate.entity.Supplier;
+import ua.okwine.productexpirationdate.service.SupplierService;
+
+import java.util.UUID;
+
+
+@Controller
+@RequestMapping("/suppliers")
+@AllArgsConstructor
+public class SupplierController {
+
+    private final SupplierService supplierService;
+
+    @GetMapping("/suppliersList")
+    public String getAll(Model model) {
+
+        model.addAttribute("suppliers", supplierService.findAll());
+
+        return "suppliers/suppliersList";
+    }
+
+    @GetMapping("/showFormForAdd")
+    public String getFormForAdd(Model model) {
+        Supplier supplier = new Supplier();
+
+        model.addAttribute("supplier", supplier);
+
+        return "suppliers/newSupplierForm";
+    }
+
+    @PostMapping("/save")
+    public String saveProvider(@ModelAttribute("provider") Supplier supplier) {
```

Comment:
Provider in ModelAttribute and Supplier name, use ctrl+shift+F to find all "provider" in your code and change it.